Documentation (#78)
hpi-bp1819-naumann/deequ: analyzerDoc.md

## ApproxCountDistinct
    Compute approximated count distinct with [HyperLogLogPlusPlus](https://en.wikipedia.org/wiki/HyperLogLog#HLL++)

> ## ApproxQuantile

Reviewer: According to wikipedia, this is not called the x-quantile, but the x-percentile. But his is also different in Amazon's code

Author: That's true but we should keep it consistent with Amazon's notation for now

    Approximate quantile analyzer. 
    Yields the x-quantile; a value, for which it holds true, that exaclty x percent of the total values are smaller than the yielded value.
    The quantile x can be passed to the analyzer as parameter, x has to be a number between 0 and 1. Choosing 0.5 will yield the median.
    The allowed relative error compared to the exact quantile can be configured with `relativeError` parameter. A `relativeError` = 0.0 would yield the exact quantile while increasing the computational load.
## ApproxQuantiles
    The same as ApproxQuantile, can be given a sequence of quantiles.
## Completeness

>     Completeness is the fraction of the number of non-null values devided by the number of all values in a column.

Reviewer: ```suggestion
    Completeness is the fraction of the number of non-null values divided by the number of all values in a column.
```

## Compliance

>     Compliance is a measure of the fraction of rows that complies with the given column constraint.

Reviewer: ```suggestion
    Compliance is a measure of the fraction of rows that complies with the given constraint. This constraint may span multiple columns.
```

>     E.g if the constraint is "att1>3" and data frame has 5 rows with att1 column value greater than

Reviewer: ```suggestion
    e.g. Given the constraint is "att1 > 3" and the data frame has 5 rows with an att1 column value greater than
```

>     3 and 10 rows under 3; a DoubleMetric would be returned with 0.33 value

Reviewer: ```suggestion
    3 and 10 rows with a value lower than 3, a DoubleMetric of 0.33 would be returned.
```

## Correlation
    Computes the [pearson correlation coefficient](https://en.wikipedia.org/wiki/Pearson_correlation_coefficient) between the two given columns.

>     It has a value between -1 and +1, where +1 is total positive linear correlation, 0 is no linear correlation, and −1 is total negative linear correlation.

Reviewer: ```suggestion
    It returns a value between -1 and +1, where +1 is total positive linear correlation, 0 is no linear correlation, and −1 is total negative linear correlation.
```

## CountDistinct
    Number of distinct values in the column.
## DataType
    Yields a distribution map, including the overall number of values of each datatype and the percentage of each datatype. 

>     Possible datatypes are Boolean, Fractional, Integral, Unknown and String.

Reviewer: ```suggestion
    Possible datatypes are Boolean, Fractional, Integral, and String. If the type can't be determined, Unknown is returned.
```

## Distinctness
    Distinctness is the fraction of the number of distinct values devided by the number of all values of a column. Can be given a sequence of columns.
## Entropy

>     [Entropy](https://en.wikipedia.org/wiki/Entropy_(information_theory)) is a measure of the level of information contained in a message. Given the probability distribution over values in a column, it describes how many bits are required to identify a value.

Reviewer: ```suggestion
    [Entropy](https://en.wikipedia.org/wiki/Entropy_(information_theory)) is a measure of the level of information contained in a message. Given the probability distribution over values in a column, it describes how many bits are required to identify a value. A highly diverse column will result in high entropy. Few different values will result in low entropy.
```

## Histogram

>     Divides the columns of the values in the given column into buckets. The number of buckets has to be given by the user and has to be smaller than or euqal to 1000. Furthermore the user has to give a user defined function as an input parameter that is applied to the bins.

Reviewer: I don't get what the user defined function is supposed to be.

>     Divides the columns of the values in the given column into buckets. The number of buckets has to be given by the user and has to be smaller than or euqal to 1000. Furthermore the user has to give a user defined function as an input parameter that is applied to the bins.

Reviewer: ```suggestion
    Computes the histogram for the given columns. Small buckets may be filtered in the output. Furthermore the user has to specify a user defined function as an input parameter that is applied to the bins.
```

## Maximum

>     Returns the largest value in the given column. Works only for numerical columns. Only for numeric columns.

Reviewer: ```suggestion
    Returns the largest value in the given column. Only for numeric columns.
```

## Mean

>     This is the average over all values of the given column. Therefore it is the fraction of the Sum and the Size of the column. Only for numeric columns.

Reviewer: ```suggestion
    This is the average over all values of the given column. Therefore it is the fraction of the sum and the size of the column. Only for numeric columns.
```

## Median

>     Returns the value in the middle of the given column. This is more robust against outliers than the mean. It is equal to the 2-quantile.

Reviewer: ```suggestion
    For n rows returns the value with index n/2 in the specified column if sorted. If n is odd the mean between the surrounding indices is computed. This is more robust against outliers than the mean. It is equal to the 2-quantile.
```

Reviewer: Would also be 2-percentile according to wikipedia

## Minimum
    Returns the smallest value in the given column. Only for numeric columns.
## Mode

>     This is the most common value in the given column.

Reviewer: ```suggestion
    This is the most common value in the given column. Not necessarily unique.
```

## MutualInformation

>     It measures how much having information about one column reveals about aonther. If both columns are independent, the mutual information is zero. If one of the columns determines all values of the other column and the other way around, there is a functional dependency between the columns. In that case, the mutual information is equal to either of both columns' Entropy.

Reviewer: ```suggestion
    Measure of how much information one column reveals about another. If both columns are independent, the mutual information is zero. If one of the columns determines all values of the other column and the other way around, there is a functional dependency between the columns. In that case, the mutual information is equal to either of both columns' entropy.
```
